src/modules/claim/claim.service.ts

```diff
@@ -280,6 +280,46 @@ export class ClaimService {
     return qb.getMany();
   }
 
+  /**
+   * Get claims issued for given subject
+   * @param did subjects's DID
+   * @param filters additional filters
+   */
+  async getBySubject({
+    subject,
+    filters = {},
+    currentUser,
+  }: {
+    subject: string;
+    filters?: QueryFilters;
+    currentUser?: string;
+  }) {
+    const { isAccepted, parentNamespace } = this.parseFilters(filters);
+    const qb = this.claimRepository
+      .createQueryBuilder();
+
+    if (isAccepted !== undefined) {
+      qb.andWhere('"isAccepted" = :isAccepted', {
+        isAccepted,
+      });
+    }
+
+    if (parentNamespace) {
+      qb.andWhere('"parentNamespace" = :namespace', {
+        namespace: parentNamespace,
+      });
+    }
+
+    if (currentUser) {
+      qb.andWhere(new Brackets(query => {
+        query.where(':currentUser = ANY ("claimIssuer")', {
+          currentUser,
+        }).orWhere(':currentUser = requester', { currentUser })
+      }))
+    }
+    return (await qb.getMany()).filter((claim) => claim.subject === subject);
```

Comment:
IMO, I would add a comment here (maybe a TODO comment) mentioning that this should be a part of the database query (as I think you explained in the PR comment)

Reply:
The problem is that subject doesn't stored in database. It is extracted from token, when claim loaded from db